lib/stella/helpers.rb

```diff
@@ -0,0 +1,103 @@
+require 'typhoeus/adapters/faraday'
+
+module Stella
+  module Helpers
+    extend ActiveSupport::Concern
+
+    @@types = []
+
+    included do
+      if defined? Rails
+        env = Rails.env
+      else
+        env = 'test'
+      end
+
+      index_name [search_index_name, env].join('_')
+
+      after_save :es_index
+      after_destroy :es_delete
+
+      attr_accessor :es_indexing
+
+      @@types << self
+    end
+
+    # track dependent classes for spec support
+    def self.types
+      @@types
+    end
+
+    def es_index
+      self.es_indexing = true
+      __elasticsearch__.index_document
+    ensure
+      self.es_indexing = nil
+    end
+
+    def es_delete
+      es_delete_document id
+    end
+
+    def es_transform
+      { index: { _id: id.to_s, data: as_indexed_json } }
+    end
+
+    module ClassMethods
+      ## Searching
+
+      def stella_raw_search(params = {})
+        search(stella_query(params))
+      end
+
+      # @return an array of database records mapped using an adapter
+      def stella_search(params = {})
+        rsp = stella_raw_search(params)
+        params[:raw] ? rsp.response : rsp.records.to_a
+      end
+
+      ## Indexing
+
+      # default index naming scheme is pluralized model_name
+      def search_index_name
+        model_name.route_key
+      end
+
+      def batch_to_bulk(batch_of_ids)
+        find(batch_of_ids).map(&:es_transform)
+      end
+
+      def bulk_index(batch_of_ids)
+        __elasticsearch__.client = Elasticsearch::Client.new url: ENV['ELASTICSEARCH_URL']
```

Comment:
Hmm. There's an argument for keeping client configuration out of the library altogether. I'm not sure why we need this line in any case - this could be legacy code left over from our cluster migration period.